runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCacheCursorFactory.java

```diff
@@ -962,6 +963,93 @@ public String toString()
             }
         }
 
+        private static final class EagerOr extends KafkaFilterCondition
```

Comment:
As discussed, suggest creating a superclass `Or` and subsclasses `LazyOr` and `EagerOr` for improved readability.

Only the `.test(...)` method implementation will differ between `LazyOr` and `EagerOr`, so everything else can be implemented in the `Or` superclass.